### PR TITLE
[VWIP] Python overhaul

### PR DESCRIPTION
**Pull request by teonbrooks**
This is the initial attempts to gut out the dependency on a local Python kernel and the communication between the app and kernel. Pyodide offers a full working kernel implemented in WebAssembly. This PR attempts to identify pain points, necessary patches for upstream dependencies, and testing ground for change.

### CODE REVIEW

**Comment by teonbrooks** [comment]
you can follow along with the wishlist of things for brainwaves x pyodide [here](https://definitely-staging.iodide.io/notebooks/19/).

File: app/components/HomeComponent/index.js [hunk 1]
```diff
@@ -16,16 +16,18 @@ import {
 import InputModal from '../InputModal';
 import SecondaryNavComponent from '../SecondaryNavComponent';
 import OverviewComponent from './OverviewComponent';
+import { languagePluginLoader } from "../../utils/pyodide/pyodide";
+
+// this initiates pyodide
+languagePluginLoader;
```
Comment (teonbrooks) [inline app/components/HomeComponent/index.js]: Would probably move this to a startup epic that matches how the jupyter kernel is initiated on app load

**Comment by jdpigeon** [comment]
Does this PR deprecate #74?

**Comment by teonbrooks** [comment]
yes. closes #74.

#74 was done to show pyodide working without breaking anything.

File: app/epics/pyodideEpics.js [hunk 2]
```diff
@@ -120,15 +119,22 @@ const receiveStream = payload => ({
...
-const launchEpic = (action$, state$) =>
+const launchEpic = action$ =>
   action$.ofType(LAUNCH).pipe(
-    mergeMap(languagePluginLoader),
-    tap(() => console.log('launched pyodide'))
+    tap(() => console.log("launching")),
+    mergeMap(async () => {
+      await languagePluginLoader;
+      console.log("loaded language plugin");
+      // using window.pyodide instead of pyodide to get linter to stop yelling ;)
```
Comment (teonbrooks) [inline app/epics/pyodideEpics.js]: lol

**Comment by jdpigeon** [comment]
I think we'll make this PR the place for a complete Jupyter replacement. Hopefully can find some time to get to the matplotlib integration step soon.

**Comment by teonbrooks** [comment]
cool. getting a matplotlib plot to work should be straightforward. there is just a little bit we would need to tweak but I have it documented. 
getting the epochs viewer will be a bit more work. i think it would be a matter of getting some event listeners in place.
happy to hack on this with you, should be around town for a bit

**Comment by teonbrooks** [comment]
@jdpigeon should we make a to-do list of things we have left to address here?

**Comment by teonbrooks** [comment]
At this point we have been able to handle the follow:
- go from loading raw csv -> making raw object -> filtering -> epochs

We have the following left:
- plot the interactive epoch viewer
- connect the epochs to the analyze screen
- plotting the condition differences

Some things to keep in mind:
- the loading of pyodide is a bit slow
  - this is blocking the main thread, which it should not be
- the import might be slowed down with the loading of Pandas in addition to SciPy
- we might want to mount the virtual filesystem to the real filesystem so we can have direct read and write from the app via Pyodide (currently restricted to the js side)

**Comment by teonbrooks** [comment]
closes https://github.com/makebrainwaves/BrainWaves/issues/43

**Comment by teonbrooks** [comment]
hey @jdpigeon, could you rebase this branch to the latest master and I will then cherry-pick some of the commits from my local branch to get it all up-to-date

**Comment by teonbrooks** [comment]
stumbled upon this, which might be useful:
https://pyodide.readthedocs.io/en/stable/using_pyodide_from_webworker.html
https://github.com/iodide-project/pyodide/blob/master/src/webworker.js

since the data processing is happening in python world and js only need images, maybe those can be passed with the onMessage. it will take python off the main thread and speed up the app.

**Comment by jdpigeon** [comment]
> stumbled upon this, which might be useful:
> https://pyodide.readthedocs.io/en/stable/using_pyodide_from_webworker.html
> https://github.com/iodide-project/pyodide/blob/master/src/webworker.js
> 
> since the data processing is happening in python world and js only need images, maybe those can be passed with the onMessage. it will take python off the main thread and speed up the app.

That sounds brilliant. Thanks Teon

**Comment by jdpigeon** [comment]
Moving this branch from teonbrooks to makebrainwaves. Should be easier to bring through the home stretch that way.

**Pull request merged**